### PR TITLE
[metasrv] move protobuf files out of metasrv

### LINKED ISSUE #3528: The first thing to do is to move `metasrv/meta.proto` out and put it into a new crate so that other crate is able to access metasrv without depending on it.
- [ ] The first thing to do is to move `metasrv/meta.proto` out and put it into a new crate so that other crate is able to access metasrv without depending on it.

_Originally posted by @drmingdrmer in https://github.com/datafuselabs/databend/issues/3515#issuecomment-996512151_

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Move protobuf files out of metasrv so that other crates can access metasrv without depending on it.

## Changelog

- Improvement

## Related Issues

Fixes #3528 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: .licenserc.yaml [hunk 1]
```diff
@@ -27,9 +27,7 @@ header:
     - "**/*.sql"
     - "rust-toolchain"
     - "rustfmt.toml"
-    - "**/*.proto"
```
Comment (drmingdrmer) [inline .licenserc.yaml:30]: Hmm... why remove these ignores? 🤔

File: common/meta/raft-store/build.rs [hunk 2]
```diff
@@ -17,16 +17,15 @@
...
 }
 
 fn build_proto() {
     let manifest_dir =
         env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR env variable unset");
 
-    // Changing OUT_DIR fails docker build because src/proto is read-only.
-    // env::set_var("OUT_DIR", "src/proto");
+    // Changing OUT_DIR fails docker build because src/protobuf is read-only.
+    // env::set_var("OUT_DIR", "src/protobuf");
```
Comment (drmingdrmer) [inline common/meta/raft-store/build.rs:28]: These 2 lines can be just removed

File: common/meta/raft-store/build.rs [hunk 3]
```diff
@@ -17,16 +17,15 @@
 use std::env;
 use std::path::Path;
 fn main() {
-    common_building::setup();
```
Comment (drmingdrmer) [inline common/meta/raft-store/build.rs:20]: Isn't the proto gonna moved into a standalone crate? 

The proto is about RPC protocol. IMHO, it has nothing to do with raft-store. Unless I missed anything. 🤔

Comment (ariesdevil) [reply common/meta/raft-store/build.rs:20]: Cause I encounter the rust orphan rule :(

Comment (flaneur2020) [reply common/meta/raft-store/build.rs:20]: > The proto is about RPC protocol. IMHO, it has nothing to do with raft-store. Unless I missed anything. 🤔

+1

how about calling it meta-proto or meta-api? q.q

**Comment by Xuanwo** [comment]
Oh, the workaround does work.

**Comment by ariesdevil** [comment]
> Oh, the workaround does work.

Thx a lot~

**Pull request merged**